task/mgmt-gateway/src/mgs_gimlet.rs

```diff
@@ -268,11 +303,13 @@ impl SpHandler for MgsHandler {
             component
         }));
 
-        match component {
-            SpComponent::SP_ITSELF => Ok(self.common.status()),
+        let status = match component {
+            SpComponent::SP_ITSELF => self.sp_update.status(),
```

Comment:
No, but it's not obvious why that is. I'll add comments to explain. A first attempt to do so here:

MGS hides the fact that aux flash images exist from its clients - they only know about updates. When they ask for the status of or to abort an update, they will say "give me the status of the update being applied to `SP_ITSELF`" - at that level, we don't actually apply updates to `SP_AUX_FLASH`.

`update_chunk`, on the other hand, is only called directly by MGS as a part of delivering the update data, never by a client that doesn't know the details. Those chunks can include the detail of whether the data in that chunk is for the SP or for the aux flash. Strictly speaking even that isn't necessary, since the contract could be "the first `aux_flash_size` bytes are for the aux flash and the remaining `sp_image_size` bytes are for the SP", but it seemed slightly safer to include the component name along with the data.

Reply:
Added comments to this and update_abort in 644a64a3